kubernetes/resource_kubernetes_service_test.go

```diff
@@ -391,27 +393,31 @@ func TestAccKubernetesService_nodePort(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "spec.0.port.0.port", "10222"),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.port.0.protocol", "TCP"),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.port.0.target_port", "22"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.port.0.app_protocol", appProtocolFirst),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.port.1.name", "second"),
 					resource.TestCheckResourceAttrSet(resourceName, "spec.0.port.1.node_port"),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.port.1.port", "10333"),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.port.1.protocol", "TCP"),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.port.1.target_port", "33"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.port.1.app_protocol", appProtocolSecond),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.selector.%", "1"),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.selector.App", "MyApp"),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.session_affinity", "ClientIP"),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.type", "NodePort"),
 					testAccCheckServicePorts(&conf, []api.ServicePort{
 						{
-							Name:       "first",
-							Port:       int32(10222),
-							Protocol:   api.ProtocolTCP,
-							TargetPort: intstr.FromInt(22),
+							AppProtocol: &appProtocolFirst,
```

Comment:
There is a helper function [ptrToString](https://github.com/hashicorp/terraform-provider-kubernetes/blob/6db39254a8ce8b2b8cf885b6efc648e7f8821fce/kubernetes/structures.go#L206) if you didn't want to have to declare a variable and take the address of it.